src/cswinrt/code_writers.h

```diff
@@ -7219,7 +7285,29 @@ CopyToUnmanaged_%(null, __%, __%Size, (void**)%);
                     m.local_type = m.param_type;
                     m.interface_guid = w.write_temp("%", bind<write_iid_guid>(type));
                     m.marshal_by_object_reference_value = true;
-                    if (distance(type.GenericParam()) > 0)
+                    if (is_projected_as_nullable(type))
+                    {
+                        m.is_boxed_value = true;
+                        auto generic_arg = w.get_generic_arg(0);
+                        call(generic_arg,
```

Comment:
This logic to get the right marshaller seems the same above, I wonder if it should be some shared helper method?

Reply:
It is basically the same right now for this.  I did think about moving this elsewhere but kept it separate for now as we do the managed marshaler and abi marshaler decisions separately today.